test/es6/es6IsConcatSpreadable.js

```diff
@@ -124,7 +125,515 @@ var tests = [
+            }
+
+            test(Number, 0, [1, 2, 3]);
+            test(Number, -0.1, [1.1, 2.2, 3.3]);
+            test(Number, NaN, ["a", "b", "c"]);
+            test(Number, 321, [1, "ab", 2.2, 2, NaN, 3, function(){ }]);
+
+            test(Boolean, true, [1, 2, 3]);
+            test(Boolean, false, [1.1, 2.2, 3.3]);
+            test(Boolean, true, ["a", "b", "c"]);
+            test(Boolean, false, [1, "ab", 2.2, 2, NaN, 3, function(){ }]);
+
+            test(RegExp, /^/, [1, 2, 3]);
+            test(RegExp, /abc/, [1.1, 2.2, 3.3]);
+            test(RegExp, /(\d+)/, ["a", "b", "c"]);
+            test(RegExp, /^\s*\S+\s*$/, [1, "ab", 2.2, 2, NaN, 3, function(){ }]);
+        }
+    },
+    {
+        name: "spread String object",
+        body: function ()
+        {
+            function test() {
+                var s = new String("abc");
+                areEqual([s], [].concat(s), "string");
+
+                s[Symbol.isConcatSpreadable] = true;
+                areEqual(['a', 'b', 'c'], [].concat(s), "string s[Symbol.isConcatSpreadable] == true");
+
+                s[Symbol.isConcatSpreadable] = false;
+                areEqual([s], [].concat(s), "string s[Symbol.isConcatSpreadable] == false");
+
+                s[Symbol.isConcatSpreadable] = true;
+                areEqual(['a', 'b', 'c'], [].concat(s), "string s[Symbol.isConcatSpreadable] == true");
+
+                delete s[Symbol.isConcatSpreadable];
+                areEqual([s], [].concat(s), "string s[Symbol.isConcatSpreadable] deleted");
+
+                String.prototype[Symbol.isConcatSpreadable] = true;
+                areEqual(['a', 'b', 'c'], [].concat(s), "string.prototype[Symbol.isConcatSpreadable] == true");
+
+                String.prototype[Symbol.isConcatSpreadable] = false;
+                areEqual([s], [].concat(s), "string.prototype[Symbol.isConcatSpreadable] == false");
+
+                String.prototype[Symbol.isConcatSpreadable] = true;
+                areEqual(['a', 'b', 'c'], [].concat(s), "string.prototype[Symbol.isConcatSpreadable] == true");
+
+                delete String.prototype[Symbol.isConcatSpreadable];
+                areEqual([s], [].concat(s), "string.prototype[Symbol.isConcatSpreadable] deleted");
+            }
+
+            test();
+        }
+    },
+    {
+        name: "Revokable proxy revoked when retrieving [@@isConcatSpreadable]",
+        body: function ()
+        {
+            // proxy revoked
+            var obj = {};
+            var pobj = Proxy.revocable(obj, {
+                get: function(target, prop) {
+                    if (prop === Symbol.isConcatSpreadable) { pobj.revoke(); }
+                    return obj[prop];
+                }
+            });
+            assert.throws(()=>[].concat(pobj.proxy), TypeError, 'proxy revoked', 'method  is called on a revoked Proxy object');
+        }
+    },
+    {
+        name: "[@@isConcatSpreadable] getter altering binding",

```

Comment:
Added in latest rev.

---
In reply to: [68466415](https://github.com/Microsoft/ChakraCore/pull/1198#discussion_r68466415) [](ancestors = 68466415)